specs/Menu.md

```diff
@@ -1,731 +1,540 @@
+### SubMenu
 
-By default Focus should go to the tab stops and it's up to each menuItem to decide if it has tabindex=0. An example of a menuItem that shouldn't be focusable would be a Divider.
+Creates a `Menu` component with `MenuItem` trigger and handles the positioning of the nested menu.
 
-However for the menu we should provide a focus wrapper that is able to track focus using role="menuitem" to determine what item should get focused.
+## Sample code
 
-The menu does not need to provide a native way to be controlled without getting focus directly. Like above, there should be a simple utility wrapper that could be written to implement this.
+The below samples do not represent the definitive props of the final implemented component, but represent the ideal final implementations. Can be subject to change during the implementation phase.
 
-##### On Menu Click
+### Default Menu
 
-Some implementations of menu have a general `onClick` function that is applied to all items as a way of getting more information.
+```typescript
+const trigger = <button> Open menu </button>
 
-Recommendation: There is not a need to have an `onClick` that is called whenever an item is clicked. Instead the root `menu` should take an `onClick` that will fire if a child's on click does not prevent default.
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem title="Option 1" />
+    <MenuItem title="Option 2" />
+    <MenuItem title="Option 3" />
+  <Menu>
+)
+```
 
-#### Menu Items
+### Sections and submenus
+
+```typescript
+const trigger = <button> Open menu </button>
+
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem title="Option 1" />
+    <MenuSection title="Section">
+      <MenuItem title="Section Option 1">
+      <MenuItem title="Section Option 2">
+      <MenuItem title="Section Option 3">
+    <MenuSection />
+    <SubMenu title="Open submenu">
+      <MenuItem title="Submenu Option 1">
+      <MenuItem title="Submenu Option 2">
+      <MenuItem title="Submenu Option 3">
+    <SubMenu>
+  <Menu>
+)
+```
 
-##### On Item Click
+### Standlone
 
-The on item click should be supplied individually by the consuming component
+```typescript
+const [open] = React.useState(false);
 
-##### Focus
+const menu = (
+  <CustomSurface open={open}>
+    <MenuList>
+      <MenuItem title="Option 1" />
+      <MenuItem title="Option 2" />
+      <MenuItem title="Option 3" />
+    <MenuList>
+  <CustomSurface>
+)
+```
 
-The menu item should change border and background color on focus.
+### Selection
+
+```typescript
+const trigger = <button> Open menu </button>
+const [selectedItems, setSelectedItems] = React.useState([]);
+
+// basic checkbox example
+const menuCheckbox = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuItem kind="checkbox" key={1} title="Option 1">
+    <MenuItem kind="checkbox" key={2} title="Option 2">
+    <MenuItem kind="checkbox" key={3} title="Option 3">
```

Comment:
`index` in the prototype is used intentionally as `key` is not a React prop and can't be accessed inside a component

Reply:
Changed for `index` might play around with the prototype and have a think. But as I keep stating, this spec does not set the implementation in stone